pkg/daemon/ceph/osd/daemon.go

```diff
@@ -144,6 +150,34 @@ func RunFilestoreOnDevice(context *clusterd.Context, mountSourcePath, mountPath
 	// unmount the device before exit
 	defer sys.UnmountDevice(mountPath, context.Executor)
 
+	// chown the mounted path as ceph:ceph
+	logger.Debugf("chown-ing mount path %q as %q", mountPath, "ceph:ceph")
+	g, err := user.LookupGroup("ceph")
+	if err != nil {
+		return fmt.Errorf("failed to determine group id of group %q. %+v", "ceph", err)
+	}
+	gid, err := strconv.Atoi(g.Gid)
+	if err != nil {
+		return fmt.Errorf("failed to convert group id %q of group %q to an int. %+v", g.Gid, "ceph", err)
+	}
+	u, err := user.Lookup("ceph")
+	if err != nil {
+		return fmt.Errorf("failed to determine user id of user %q. %+v", "ceph", err)
+	}
+	uid, err := strconv.Atoi(u.Uid)
+	if err != nil {
+		return fmt.Errorf("failed to convert user id %q of user %q to an int. %+v", u.Uid, "ceph", err)
+	}
+	err = filepath.Walk(mountPath, func(name string, info os.FileInfo, err error) error {
+		if err == nil {
+			return os.Chown(name, uid, gid)
```

Comment:
I wouldn't chown on filestore. When users have thousands (millions?) of objects and large directory trees this would take forever to complete.

If the concern is around the wrong ownership, we can run the process with `--setuser-match-path`, just like https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/osd/spec.go#L247

Reply:
Does this work in non-ceph-volume cases? The `chown` was already happening before my work here but by using `chown` from an init container instead of using Go.

Reply:
Yes, if we are talking about chowing `/var/lib/ceph/osd/osd-$ID/` on Filestore then yes we should not do that.

Reply:
I think I might be able to alter the `sys.MountDevice` to use the option `umask=filePermissions,gid=ownerGroupID,uid=ownerID`?

Reply:
Yes!

Reply:
The mount kept failing with those options, but we can still use `--setupser-match-path` as suggested.